### PR TITLE
Tailor pipeline yaml to 2.0-preview

### PR DESCRIPTION
**Pull request by aeunms**
After the yaml is merged in, I will point our pipeline to this file and have it run the content.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -176,71 +175,92 @@ steps:
...
+    condition: and(
+      in(variables['Agent.JobStatus'], 'Succeeded', 'SucceededWithIssues'),
+      startsWith(variables['Build.SourceBranch'], 'refs/tags/v'))
+    inputs:
+      ConnectionType: 'AzureRM'
+      azureSubscription: '$(AzureSubscription)'
+      appType: 'webApp'
+      WebAppName: 'musicalsink'
+      package: '$(Build.ArtifactStagingDirectory)\teams-test-app\$(Build.BuildId).zip'
+    displayName: 'Deploy to musical sink'
```
Comment (erinha) [inline azure-pipelines.yml:190]: nit: I think the displayName should reference the webappname, i.e., `'Deploy to musicalsink'`. Same for `'Deploy to cloudroll'`

File: azure-pipelines.yml [hunk 2]
```diff
@@ -75,81 +100,55 @@ steps:
...
-    condition: and(in(variables['Agent.JobStatus'], 'Succeeded', 'SucceededWithIssues'), eq(variables['Build.SourceBranch'], 'refs/heads/develop'))
-    inputs:
-      ConnectionType: 'AzureRM'
-      azureSubscription: '$(AzureSubscription)'
-      appType: 'webApp'
-      WebAppName: 'cloudroll'
-      package: '$(Build.ArtifactStagingDirectory)\teams-test-app\$(Build.BuildId).zip'
-    displayName: 'Deploy to cloud roll'
-
-  - task: ESLint@1
```
Comment (murthys) [inline azure-pipelines.yml:132]: Why are we removing these tasks

Comment (aeunms) [reply azure-pipelines.yml:132]: We're not, they've just been edited and reordered and the GitHub UI (Or I guess also CodeFlow if you're using that) thinks they've been "deleted"

**Review by murthys — APPROVED** [review]

**Comment by erinha** [comment]
$(bundleAnalysisTask.bundleAnalysisComment)

**Comment by erinha** [comment]
$(bundleAnalysisTask.bundleAnalysisComment)

**Comment by erinha** [comment]
$(bundleAnalysisTask.bundleAnalysisComment)

**Comment by erinha** [comment]
$(bundleAnalysisTask.bundleAnalysisComment)

**Comment by erinha** [comment]
$(bundleAnalysisTask.bundleAnalysisComment)

File: azure-pipelines.yml [hunk 3]
```diff
@@ -176,98 +173,162 @@ steps:
...
+
+  - task: Yarn@3
+    condition: and(
+      eq(variables['Build.SourceBranch'], 'refs/heads/2.0-preview'),
+      in(variables['Agent.JobStatus'], 'Succeeded', 'SucceededWithIssues'),
+      ne(variables['Build.Reason'], 'PullRequest'),
+      ne(variables['Build.Reason'], 'Manual'))
+    inputs:
+      Arguments: 'publish-sdk --tag next-dev'
+    displayName: 'yarn publish-sdk (dev commit)'
```
Comment (erinha) [inline azure-pipelines.yml:235]: Should this also be commented out for now?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -176,98 +173,162 @@ steps:
...
+  
+  - powershell: |
+      $npmVer=$(node -p "require('./packages/teams-js/package.json').version") 
+      Write-Host "##vso[task.setvariable variable=Version;isOutput=true]$npmVer"
+    displayName: 'set build number'
```
Comment (erinha) [inline azure-pipelines.yml:240]: Can we update the displayName to be "Set package.Version Variable" please?

**Review by erinha — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml [hunk 5]
```diff
@@ -176,98 +173,162 @@ steps:
...
+        packages\teams-js\publish.js
+      TargetFolder: '$(Build.ArtifactStagingDirectory)\Scripts'
+      flattenFolders: true
+    displayName: 'Copy publish.js Script for NPM'
+
+  - task: PublishBuildArtifacts@1
+    inputs:
+      PathtoPublish: '$(Build.ArtifactStagingDirectory)\CDNFeed'
+      ArtifactName: 'CDNFeed'
+    displayName: 'CDN feed to build Artifacts'
```
Comment (erinha) [inline azure-pipelines.yml:280]: Can displayName be `Publish CDN feed to build Artifacts`?

File: azure-pipelines.yml [hunk 6]
```diff
@@ -176,98 +173,162 @@ steps:
...
+  
   - task: PublishBuildArtifacts@1
     inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)\$(prepNextRelease.NextVersion)'
-      ArtifactName: '$(prepNextRelease.NextVersion)'
-    displayName: 'Publish TeamsJS to build Artifacts'
+      PathtoPublish: '$(Build.ArtifactStagingDirectory)\NPMFeed'
+      ArtifactName: 'NPMFeed'
+    displayName: 'NPM feed to build Artifacts'
```
Comment (erinha) [inline azure-pipelines.yml:286]: Can displayName be 'Publish NPM feed to Build Artifacts'?

File: azure-pipelines.yml [hunk 7]
```diff
@@ -176,98 +173,162 @@ steps:
...
+  
+  - task: PublishBuildArtifacts@1
+    inputs:
+      PathtoPublish: '$(Build.ArtifactStagingDirectory)\Scripts'
+      ArtifactName: 'Scripts'
+    displayName: 'Scripts to build Artifacts'
```
Comment (erinha) [inline azure-pipelines.yml:292]: Can displayName be 'Publish Scripts to Build Artifacts'?

**Review by erinha — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml [hunk 8]
```diff
@@ -176,98 +173,162 @@ steps:
...
+      SourcePath: '$(Build.ArtifactStagingDirectory)\CDNFeed\$(package.Version)'
+      ConnectionType: 'EnvironmentVariables'
+      Environment: 'PublicCloud'
+      ContainerName: 'teams-js'
+      AdditionalArgumentsForBlobCopy: '--overwrite=false --log-level=INFO --recursive'
+    env:
+      M365CDN_APP_ID: $(M365-CDN-CLIENT-ID-PROD)
+      M365CDN_APP_KEY: $(M365-CDN-CLIENT-SECRET-PROD)
+      M365CDN_TENANT_ID: $(M365-APP-TENNANT-ID)
+    displayName: 'Push TeamsJS to M365 1CDN(Prod) for (Main)'
```
Comment (erinha) [inline azure-pipelines.yml:334]: Can we change this to 'Push TeamsJS to M365 1CDN (Prod) for (2.0-preview release)'?

**Review by erinha — APPROVED** [review]
Looks good!

File: packages/teams-js/publish.js [hunk 9]
```diff
@@ -1,18 +1,17 @@
 const fs = require('fs-jetpack');
 const cp = require("child_process");
 const npmRegistry = 'https://registry.npmjs.org';
+const sdkPath = 'packages/teams-js';
```
Comment (erinha) [inline packages/teams-js/publish.js:4]: Remove since the file is in this directory.

File: azure-pipelines.yml [hunk 10]
```diff
@@ -176,98 +173,162 @@ steps:
...
       ne(variables['Build.Reason'], 'PullRequest'),
-      ne(variables['Build.Reason'], 'Manual')
-      )
+      ne(variables['Build.Reason'], 'Manual'))
     inputs:
-      Arguments: 'publish-sdk --tag next'
-    displayName: 'yarn publish-sdk (develop)'
+      filename: node
+      arguments: publish.js
+      workingFolder: '$(System.DefaultWorkingDirectory)'
```
Comment (erinha) [inline azure-pipelines.yml:212]: The workingFolder should be '$(System.DefaultWorkingDirectory)\packages\teams-js'

**Review by erinha — CHANGES_REQUESTED** [review]

**Review by erinha — APPROVED** [review]
Looks good!

**Pull request merged**